crates/flowctl/src/collection/mod.rs

```diff
@@ -17,75 +17,39 @@ pub struct CollectionJournalSelector {
     /// The full name of the Flow collection
     #[clap(long)]
     pub collection: String,
-    /// Selects a logical partition to include. Partition selectors must be provided in the format
-    /// `<name>=<value>`. For example: `--include-partition userRegion=eu` would only include the
-    /// logical partition of the `userRegion` field with the value `eu`. This argument may be
-    /// provided multiple times to include multiple partitions. If this argument is provided, then
-    /// any other logical partitions will be excluded unless explicitly included here.
-    #[clap(long = "include-partition")]
-    pub include_partitions: Vec<Partition>,
-    /// Selects a logical partition to exclude. The syntax is the same as for `--include-partition`.
-    /// If this argument is provided, then all partitions will be implicitly included unless
-    /// explicitly excluded here.
-    #[clap(long = "exclude-partition")]
-    pub exclude_partitions: Vec<Partition>,
+    /// Selects a subset of collection partitions using the given selector.
+    /// The selector is provided as JSON matching the same shape that's used
+    /// in Flow catalog specs. For example:
+    /// '{"include": {"myField1":["value1", "value2"]}}'
+    #[clap(
+        long,
+        value_parser(parse_partition_selector),
+        conflicts_with_all(&["include-partition", "exclude-partition"])
+    )]
+    pub partitions: Option<models::PartitionSelector>,
+
+    /// Deprecated, use --partitions instead
```

Comment:
If we're just gonna error out anyway, seems like we could just remove it altogether?

Reply:
🤷 I know there's at least one person using it, so I figured why not give folks a better error message. We can always remove it the next time we're in there